File: snake/server/app/src/main/java/server/messages/ServerRoomIdMessage.java
package server.messages;

import java.time.LocalDateTime;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.JsonSyntaxException;

public class ServerRoomIdMessage implements Message {
    private String messageCode;
    private String roomId;
    private LocalDateTime timeStamp;

    public String getMessageCode() {
        return messageCode;
    }

    public void setMessageCode(String messageCode) {
        this.messageCode = messageCode;
    }

    public String getRoomId() {
        return roomId;
    }

    public void setRoomId(String roomId) {
        this.roomId = roomId;
    }

    public LocalDateTime getTimeStamp() {
        return timeStamp;
    }

    public void setTimeStamp(LocalDateTime timeStamp) {
        this.timeStamp = timeStamp;
    }

    public String toJson() {
        Gson gson = new GsonBuilder()
                .setDateFormat("dd.MM.yyyy-HH.mm.ss")
                .create();
        return gson.toJson(this);
    }

    public ServerRoomIdMessage fromJson(String json) throws JsonSyntaxException {

        Gson gson = new GsonBuilder().setDateFormat("dd.MM.yyyy-HH.mm.ss").create();
        return gson.fromJson(json, ServerRoomIdMessage.class);
    }
}

File: snake/client/scripts/messages/ServerRoomIdMessage.js
class ServerRoomIdMessage {
    constructor() {
      this.messageCode = '';
      this.roomId = '';
      this.timeStamp = null; // Use JavaScript Date object instead of LocalDateTime
    }
  
    toJson() {
      const json = {
        messageCode: this.messageCode,
        roomId: this.roomId,
        timeStamp: this.timeStamp ? TimeFormatter.formatDateTime(this.timeStamp) : null,
      };
      return JSON.stringify(json);
    }
  
    static fromJson(json) {
      const obj = JSON.parse(json);
      const message = new ServerRoomIdMessage();
      message.messageCode = obj.messageCode;
      message.roomId = obj.roomId;
      message.timeStamp = obj.timeStamp ? TimeFormatter.parseDateTime(obj.timeStamp) : null;
      return message;
    }
}